src/v/serde/parquet/schema.h

```diff
@@ -22,13 +22,24 @@
 
 namespace serde::parquet {
 
-struct bool_type {};
-struct i32_type {};
-struct i64_type {};
-struct f32_type {};
-struct f64_type {};
+struct bool_type {
+    bool operator==(const bool_type&) const = default;
+};
```

Comment:
what does it mean to have operator== without any data members (always equal/true)?

Reply:
just to generate equality for std::variant and other classes that have these as members.